AutomatedTesting/Gem/PythonTests/AWS/Windows/core/test_aws_resource_interaction.py

```diff
@@ -57,130 +88,57 @@ def setup(launcher: pytest.fixture, cdk: pytest.fixture, resource_mappings: pyte
-                                '(Script) - [S3] Get object success: Object example.txt is downloaded.'],
-                unexpected_lines=['(Script) - [S3] Head object error: No response body.',
-                                  '(Script) - [S3] Get object error: Request validation failed, output file directory doesn\'t exist.'],
-                halt_on_unexpected=True
-                )
-
-            assert result, "Expected lines weren't found."
 
-        download_path = os.path.join(download_dir, 'output.txt')
-
-        file_was_downloaded = os.path.exists(download_path)
-        # clean up the file directories.
-        if file_was_downloaded:
-            os.remove(download_path)
-        os.rmdir(download_dir)
-
-        assert file_was_downloaded, 'The expected file wasn\'t successfully downloaded'
-
-    def test_invoke_lambda(self,
-                           level: str,
-                           launcher: pytest.fixture,
-                           cdk: pytest.fixture,
-                           resource_mappings: pytest.fixture,
-                           workspace: pytest.fixture,
-                           asset_processor: pytest.fixture
-                           ):
-        """
-        Setup: Deploys the CDK.
-        Tests: Runs the test level.
-        Verification: Searches the logs for the expected output from the example lambda.
-        """
-
-        log_monitor = setup(launcher, cdk, resource_mappings, asset_processor)
-
-        launcher.args = ['+LoadLevel', level]
-        launcher.args.extend(['-rhi=null'])
-
-        with launcher.start(launch_ap=False):
-            result = log_monitor.monitor_log_for_lines(
-                expected_lines=['(Script) - [Lambda] Completed Invoke',
-                                '(Script) - [Lambda] Invoke success: {"statusCode": 200, "body": {}}'],
-                unexpected_lines=['(Script) - Request validation failed, output file miss full path.',
-                                  '(Script) - '],
-                halt_on_unexpected=True
-            )
-
-        assert result
-
-    def test_get_dynamodb_value(self,
+    @pytest.mark.parametrize('expected_lines', [
+        ['(Script) - [S3] Head object request is done',
+         '(Script) - [S3] Head object success: Object example.txt is found.',
+         '(Script) - [S3] Get object success: Object example.txt is downloaded.',
+         '(Script) - [Lambda] Completed Invoke',
+         '(Script) - [Lambda] Invoke success: {"statusCode": 200, "body": {}}',
+         '(Script) - [DynamoDB] Results finished']])
+    @pytest.mark.parametrize('unexpected_lines', [
+        ['(Script) - [S3] Head object error: No response body.',
+         '(Script) - [S3] Get object error: Request validation failed, output file directory doesn\'t exist.',
+         '(Script) - Request validation failed, output file miss full path.',
+         '(Script) - ']])
```

Comment:
the only concern is how to know which scriptcanvas fails if error happens?

Reply:
Log monitor will raise an error with the unexpected line it found in the log.